test/modules/Relayer/Program.cs

```diff
@@ -105,6 +109,11 @@ static async Task<MessageResponse> ProcessAndSendMessageAsync(Message message, o
                     return MessageResponse.Completed;
                 }
 
+                if (duplicateMessageAuditor.ShouldFilterMessage(sequenceNumber))
```

Comment:
My understanding is that there won't be race conditions here because the message handler (ln 49) will only be running one delegate at a time.